### PR TITLE
move nixobolus' modules here

### PR DESCRIPTION
**Pull request by tupakkatapa**
Moving the homestakeros and format modules here from the [nixobolus](https://github.com/ponkila/nixobolus) repository, which is going to be archived soon. The homestakeros module flake is almost exactly the same as the nixobolus flake, but:

1. removed nixosConfigurations, `buidl` package, and format modules, as these are not needed here
2. added mev-boost package, which was probably forgotten to be added by Juuso before

Juuso requested that the module be retained within its own flake. I do not fully understand this and even somewhat despise the fact that there are flakes within a flake repository, which creates a circular import. That being said, I don't really care.

I would appreciate if Juuso would explain why the homestakeros module is moved here and why the circular import, just for the record.

### CODE REVIEW

**Comment by tupakkatapa** [comment]
Tested building a host successfully from the frontend with a relative input path before changing it to circular. Apparently, the flake URL cannot contain both ref and dir at the same time. The more you know.

File: flake.nix [hunk 1]
```diff
@@ -2,15 +2,15 @@
   inputs = {
     devenv.url = "github:cachix/devenv";
     flake-parts.url = "github:hercules-ci/flake-parts";
-    nixobolus.url = "github:ponkila/nixobolus";
+    homestakeros.url = "github:ponkila/HomestakerOS\?dir=modules/homestakeros";
```
Comment (jhvst) [inline flake.nix:5]: off-topic: Why is the question mark escaped? Is this for shell scripts or something?

File: flake.nix [hunk 2]
```diff
@@ -194,7 +194,15 @@
...
-        schema = nixobolus.outputs.exports.homestakeros;
+        schema = homestakeros.outputs.exports.homestakeros;
+
+        # Format modules
+        nixosModules.isoImage = {
+          imports = [./modules/copytoram-iso.nix];
+        };
+        nixosModules.kexecTree = {
+          imports = [./modules/netboot-kexec.nix];
+        };
```
Comment (jhvst) [inline flake.nix:205]: I was thinking of having these alongside the common options in their own flake.

**Review by jhvst — COMMENTED** [review]
>Juuso requested that the module be retained within its own flake. 

A misunderstanding. I will patch this to show and experiment what I meant, but this is a good start.

File: flake.nix (hunk 1 above)
Comment (tupakkatapa) [reply flake.nix:5]: a habit, and the syntax highlighting is nice

**Comment by jhvst** [comment]
Done now.

So, the subdirectory is a flake which has the base config, and it resolves them as exported flake configs. These modules are then imported in the root flake and passed as modules. 

The upside is that the base module, as it is a flake, can be imported by other projects as such. These should include changes that we _suggest_ that others use in combination with whatever modules homestakeros provides, but are not mandatory. This makes it possible for someone to include homestakeros as a module without necessarily running a RAMdisk or isoImage, or e.g., in a GUI environment.

Coincidentally, if someone wants to use our opinionated configs without the use of homestakeros module (like me), that is also possible because they can resolve the flake from the subdirectory. 

Do give it a look @tupakkatapa and if all looks good, we can merge to main and I can update my personal configs to include the kexecTree module.

**Comment by tupakkatapa** [comment]
I do not like that you merged the `system.nix` with the `netboot-kexec.nix` to form the base. The options in the former expression do not have anything to do with applying the format. This was also the case previously, and I've already fixed this once here: https://github.com/ponkila/nixobolus/pull/41.

Other than that this looks good, and I now undestand what you are after with the separate flake. I am going to do a quick refactoring for the packages now.

**Comment by jhvst** [comment]
Notice that these are separated into their own nixosModules. You can use either one, or both.

**Comment by tupakkatapa** [comment]
Oh, I see. Sorry about that. I didn't spot the second `nixosModules` line.

**Comment by tupakkatapa** [comment]
Please redo the Blutgang commit, as I "had to" force-push. After that, this LGTM.

**Comment by tupakkatapa** [comment]
Except, what puprose does the neovim package serve?

**Comment by jhvst** [comment]
> Except, what puprose does the neovim package serve?

https://juuso.dev/blogPosts/modular-neovim/modular-neovim-with-nix.html

**Comment by jhvst** [comment]
Great stuff, please look also into the new treefmt formatter. The diff above is a good highlight on what it does.

**Pull request merged**